streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -278,33 +251,21 @@ public void run() {
             metrics.taskCommitTimeSensor);
     }
 
-    /**
-     * <pre>
-     *  - close topology
-     *  - {@link #commit()}
-     *    - flush state and producer
-     *    - write checkpoint
-     *    - commit offsets
-     * </pre>
-     */
     @Override
-    public void suspend() {
-        log.info("{} Suspending task", logPrefix);
-        closeTopology();
-        commit();
+    protected Map<TopicPartition, Long> recordCollectorOffsets() {
+        return recordCollector.offsets();
     }
 
-    /**
-     * re-initialize the task
-     */
-    public void resume() {
-        log.info("{} Resuming task", logPrefix);
-        initTopology();
+    @Override
+    protected void flushState() {
+        log.trace("{} Flushing state and producer topology", logPrefix);
+        super.flushState();
```

Comment:
same reply as above -- all those log indicate that a method got called -- thus it's not "double logging" imho -- but it's of course verbose, but for trace level should be fine